pkgs/development/compilers/circt/circt-llvm.nix

```diff
@@ -31,6 +32,14 @@
 
   outputs = [ "out" "lib" "dev" ];
 
+  # Get rid of ${extra_libdir} (which ends up containing a path to circt-llvm.dev
+  # in circt) so that we only have to remove the one fixed rpath.
+  postPatch = lib.optionalString stdenv.isDarwin ''
+    substituteInPlace llvm/llvm/cmake/modules/AddLLVM.cmake \
+      --replace-fail 'set(_install_rpath "@loader_path/../lib''${LLVM_LIBDIR_SUFFIX}" ''${extra_libdir})' \
+        'set(_install_rpath "@loader_path/../lib''${LLVM_LIBDIR_SUFFIX}")'
+  '';
```

Comment:
This (and the `-delete_rpath @loader_path/../lib` in postFixup) isn't needed for things to work, it just produces cleaner output (makes it so that there aren't any unused rpaths). Linux actually has the same problem, there's an unused $ORIGIN/../lib in the Linux version of circt's RUNPATH.